### PR TITLE
Edwin - Added 3rd toggle to bio announcement

### PR DESCRIPTION
**Pull request by EdwinLauM**
# Description

Screen Shot 2023-06-01 at 4 57 49 PM

## Mainly changes explained:

    1. Created a new component that integrates a tri state toggle switch
    2. Added the tri state toggle switch to the bio announcement section and made changes accordingly to hit the backend endpoints 

## How to test:
  **Important: Please test with the BE [PR368](https://github.com/OneCommunityGlobal/HGNRest/pull/368).**
    1. check into current branch
    2. do `npm run start:local` to run this PR locally
    4. Log in an admin/owner account
    5. go to Reports -> Weekly Summaries Report
    6. Check that the triple state bio toggle is working. Toggling the switch between the three states should update the backend and upon refreshing the saved changes should remain.
    7. Go to Reports -> Reports and choose any user. 
    8. The changes made in Weekly Summaries Report should reflect in that users report and vice versa. 
    9. Log in an non admin/owner account.
    10. Make sure the changes are also reflected on that user. 

## Videos/Screenshots of changes:

Admin/Owner account:

Screen Shot 2023-06-01 at 5 08 05 PM

Screen Shot 2023-06-01 at 5 08 30 PM

Volunteer Account:

Screen Shot 2023-06-01 at 5 08 52 PM

### CODE REVIEW

**Review by 20chen-7 — APPROVED** [review]
Function is intended, well done!
Login as admin user:
Screenshot 2023-06-01 at 9 10 06 PM
Screenshot 2023-06-01 at 9 10 29 PM
Login as vol user:
Screenshot 2023-06-01 at 9 14 05 PM
Screenshot 2023-06-01 at 9 11 33 PM

**Review by papiasharmin — CHANGES_REQUESTED** [review]
Hello, @EdwinLauM, all the functionality you have mentioned working as intended. 
posted
posted1

But I have all account with same password, may be because of that, when i toggled the button for admin it was changing manager or volunteer. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86232710/7c756a43-b875-467e-852b-d4aa6f6fd57d

**Comment by EdwinLauM** [comment]
Hey @papiasharmin, I see it happening to me as well. However, I don't think this PR introduced this bug/behavior since I tested it with the current development branch and it looks like it happens there as well. I was wondering if you might have any leads as to why this could be happening? Not too sure if it also should be addressed in a separate PR/Task

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/6a026a88-cb1a-41b8-bca5-3595f2d78e87

**Comment by EdwinLauM** [comment]
@papiasharmin Fixed!

**Review by stevenpaalz — DISMISSED** [review]
Confirmed that this and associated backend PR is working as intended.

As admin:
Screenshot 2023-06-02 at 10 47 23 AM

As volunteer:
Screenshot 2023-06-02 at 10 52 36 AM

**Review by yufeiStella — APPROVED** [review]
Great job! It worked as intended. 
When logging in as the admin, I can change the toggle successfully and it will also be reflected on the user page:
Screen Shot 2023-06-02 at 23 06 11
Screen Shot 2023-06-02 at 23 07 55
When logging the user(manager):
Screen Shot 2023-06-02 at 23 10 35

**Review by NidaZaki — DISMISSED** [review]
The PR works as intended! Good job!
Setting the bio posted from admin account
PR#866-2
The change is being reflected in report -> people -> select that user
PR#866-1
Also, in the volunteering account, the change is reflected
PR#866

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 1]
```diff
@@ -156,19 +157,19 @@ const FormattedReport = ({ summaries, weekIndex, role }) => {
...
-            switchType="bio"
-            state={isBioPosted ? false : true}
-            handleUserProfile={() => {
-              handleChangeBioPosted(userId, isBioPosted);
-              setIsBioPosted(!isBioPosted);
+          <div className="bio-toggle">
+            <ToggleSwitch 
+              switchType="bio"
+              state={bioStatus}
+              handleUserProfile={(bio) => {
```
Comment (nathanah) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:170]:
```suggestion
        <div className="bio-toggle">
          <ToggleSwitch
            switchType="bio"
            state={bioStatus}
            handleUserProfile={(bio) => {
```
Extra indent and trailing whitespace.

**Review by nathanah — APPROVED** [review]
Just formatting changes. Everything appears to work as intended.

File: src/components/Reports/PeopleReport/PeopleReport.jsx [hunk 2]
```diff
@@ -451,14 +451,16 @@ class PeopleReport extends Component {
             </h4>
             <p>End Date</p>
           </div>
-          {!this.state.isBioPosted ? (
+          {this.state.bioStatus ? (
             <div>
-              <h4>Bio {this.state.isBioPosted ? 'posted' : 'requested'}</h4>{' '}
+              <h5>Bio {this.state.bioStatus === "default" ? "not requested" : this.state.bioStatus}</h5>{' '}
```
Comment (nathanah) [inline src/components/Reports/PeopleReport/PeopleReport.jsx:456]: Old accounts that haven't been modified have their bioPosted (which gets set to this.state.bioStatus) as true or false. 
image

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 3]
```diff
@@ -180,7 +181,9 @@ const FormattedReport = ({ summaries, weekIndex, role }) => {
     return (
       <div>
         <b>Bio announcement:</b>
-        {bioPosted ? ' Posted' : ' Requested'}
+        {bioPosted === 'default' ? 'Not requested/posted' :
+         bioPosted === 'posted' ? 'posted' : 
+         'requested'}
```
Comment (nathanah) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:186]:
```suggestion
        {bioPosted === 'default' ? ' Not Requested/Posted' :
         bioPosted === 'posted' ? ' Posted' : 
         ' Requested'}
```
Formatting

**Review by nathanah — CHANGES_REQUESTED** [review]

File: src/components/Reports/PeopleReport/PeopleReport.jsx (hunk 2 above)
Comment (EdwinLauM) [reply src/components/Reports/PeopleReport/PeopleReport.jsx:456]: I made changes to the old accounts that are set to true and false. I have manually adjusted their document values to change from false -> "requested", true -> "posted".

**Review by cvtqx — APPROVED** [review]
@EdwinLauM I've tested this PR with its BE. The triple toggle works as intended. 

1. Changing toggle to posted in the admin account:
AdminToggleToPosted

2. The change is reflected in the report:
AdminToggleReflected

3. When logging in as a volunteer the status is the same as in the admin account:
VolunteerToggleRemains

Great job!

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 4]
```diff
@@ -180,7 +181,9 @@ const FormattedReport = ({ summaries, weekIndex, role }) => {
     return (
       <div>
         <b>Bio announcement:</b>
-        {bioPosted ? ' Posted' : ' Requested'}
+        {bioPosted === 'default' ? 'Not requested/posted' :
+         bioPosted === 'posted' ? 'Posted' : 
+         'Requested'}
```
Comment (nathanah) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:186]:
```suggestion
        {bioPosted === 'default' ? ' Not requested/posted' :
         bioPosted === 'posted' ? ' Posted' : 
         ' Requested'}
```
Still missing spaces.

**Review by nathanah — CHANGES_REQUESTED** [review]

**Review by papiasharmin — DISMISSED** [review]
Hello @EdwinLauM , Sorry for being late. Everything is working as intended. Even though the bug was not from this PR as you mentioned, but you have successfully solved it, Great work. Thank you.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86232710/42fb7be2-c129-44a8-834c-4e807b40b030

**Review by nathanah — APPROVED** [review]

**Review by bienzguoa — APPROVED** [review]
Functions all work as intended!

Screenshot 2023-06-07 at 6 57 00 PM

Screenshot 2023-06-07 at 6 57 55 PM

Screenshot 2023-06-07 at 6 59 41 PM
Screenshot 2023-06-07 at 6 59 11 PM

Screenshot 2023-06-07 at 7 34 57 PM

**Comment by mahamakhan** [comment]
Screenshot 2023-06-09 191042
Screenshot 2023-06-09 185505
The toggle works 3 ways and can be seen in the respective user as well.

**Review by mahamakhan — COMMENTED** [review]
Screenshot 2023-06-09 191042
Screenshot 2023-06-09 185505


The toggle works 3 ways and can be seen in the respective user as well.

**Pull request merged**